### ISSUE TITLE
Angular cannot find templates

### ISSUE THREAD

**Issue by 7sun** [pitr/angular-rails-templates#97]
I'm having a couple issues with Angular being able to find the templates I've created. I'm using angular-rails-templates with the ui router.

The templates will only load if I add "assets/" to the beginning of the templateUrl property. Without it, Angular returns a 404 not found. However, if I navigate in the browser to "localhost:3000/assets/sub- directory/file.html, the template will load.

Prepending "assets/" was an acceptable workaround in development, but the templates will not load at all  in production. I have the app hosted on heroku.

The fact that the templateUrls require "assets/" in dev has me suspicious that angular-rails-templates is not working as designed in my app. I checked for any possible name collisions with my .js files and there are none.

For reference, I have injected templates in the module:
```.module("ootd", ['ngResource', 'templates', 'ui.router'])```

I have required all the necessary js files in application.js:
```
//= require angular
//= require angular-resource
//= require angular-ui-router
//= require angular-rails-templates
//= require_tree ../templates
//= require_tree .
```

The templates live in "app/assets/templates/sub-directory/file.html"

I am using the "angular-rails" and "angular-rails-templates" gems in my gemfile.

Not sure if this is a bug or an issue with my code. Any help is appreciated.

**Comment by ajahongir** [comment]
same issue with latest version(0.2.0) of the gem. I have rollback to version 0.1.5 and everything is well.

**Comment by pitr** [comment]
@7sun can you try rolling back to 0.1.5? Did you bump this gem recently?

You should **not** add `assets/` to template url. When you do `require_tree ../templates`, what is actually served? There should be a separate `<script> tag per each template in your `assets/template/` file. In production those templates will be located in your compiled `application-HASH.js` file.

**Comment by gatperdut** [comment]
I'm running into the same issue. Templates in assets/templates/*.html.haml, and templateUrl: 'basename.html'. I'm using the default options. However, accessing http://localhost:3000/assets/basename.html or using templateUrl: 'assets/basename.html' works fine.

**Comment by 7sun** [comment]
I was able to resolve this by declaring the version of sprockets in my gem file as 2.12.3

```gem 'sprockets', '2.12.3'```

Maybe there is a compatibility issue that bundler isn't able to catch, perhaps with the angular gem or rails version. For clarification, I'm running:
ruby 2.2.0
rails 4.2.0
angularjs-rails 1.3.14
angular-rails-templates 0.2.0

**Comment by lborgav** [comment]
I having the same problem a lot of times. @7sun 's answer is the only way that works for me

**Comment by pitr** [comment]
@7sun what version of sprockets did you have this issue with?

**Comment by alex3t** [comment]
Same issue with last sprockets 3.0.3, resolved only when downgraded to 2.12

**Comment by 7sun** [comment]
@pitr I believe it was version 3.1.0 of sprockets.

**Comment by bradleyprice** [comment]
Same issue. @7sun's fix worked for me. Downgraded from sprockets 3.1.0 to 2.12.3.

**Comment by etep** [comment]
same issue, thank you @7sun

**Comment by pywebdesign** [comment]
Thanks for the temporary fix, it is a sad problem!

**Comment by bradgreens** [comment]
Will there be a version that supports sprockets 3.x? I'm including `sprockets-es6` in my application and can't downgrade sprockets to 2.12.3. Would love to use this gem but it's a dealbreaker for now.

**Comment by pywebdesign** [comment]
this gist may make some happy tough it invalidate pretty much everything that is good about this gem:

https://gist.github.com/pywebdesign/4be13c02c528e0bd7b91

**Comment by antonshevchenko** [comment]
@7sun worked for us too! Thanks!

**Comment by GamillusMathew** [comment]
I'm using following sprockets
 * sprockets (2.12.3
 * sprockets-rails (2.3.1)
Still couldn't get angular rails templates to work correctly as given  in https://thinkster.io/angular-rails/

**Comment by stewartmatheson** [comment]
I had this issue on both 0.2.0 and 0.1.5 of this gem. Rolling back to 0.1.3 fixed the problem for me. So it looks like the templates where not being added to angular's template cache. They where not being escaped and wrapped with the angular code that registers the template with the cache on page load. Instead the raw html was being pulled in to a .js file and yielding a syntax error. Do you get syntax errors when angular attempts to load your templates?

Adding /assets to the templates will work however your telling angular to go to the back end and load the template not loading form the cache as is intended by this gem.  In this case angular expects raw HTML so it has no issue parsing the response. Has a bug been introduced post 0.1.3 that prevents html templates from being wrapped in the javascript  code angular expects. This also explains the issue of templates not working in production. If the code can't be wrapped in javascript sprockets won't be able to package it up during pre compile.

**Comment by pitr** [comment]
Fixed by #126. Version 1.0.0.beta1 is out and supports Rails 4.2+ and Sprockets 3, I'll release version 1.0.0 once we are stable.
